doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst

```diff
@@ -0,0 +1,883 @@
+    #endif
+    #if __has_include(<tf2_eigen/tf2_eigen.hpp>)
+    #include <tf2_eigen/tf2_eigen.hpp>
+    #else
+    #include <tf2_eigen/tf2_eigen.h>
+    #endif
+
+    static const rclcpp::Logger LOGGER = rclcpp::get_logger("mtc_tutorial");
+
+    class MTCTaskNode
+    {
+    public:
+    MTCTaskNode(const rclcpp::NodeOptions& options);
+
+    rclcpp::node_interfaces::NodeBaseInterface::SharedPtr getNodeBaseInterface();
+
+    void doTask();
+
+    void setupPlanningScene();
+
+    private:
+    // Compose an MTC task from a series of stages.
+    moveit::task_constructor::Task createTask();
+    moveit::task_constructor::Task task_;
+    rclcpp::Node::SharedPtr node_;
+    };
+
+    rclcpp::node_interfaces::NodeBaseInterface::SharedPtr MTCTaskNode::getNodeBaseInterface()
+    {
+    return node_->get_node_base_interface();
+    }
+
+    MTCTaskNode::MTCTaskNode(const rclcpp::NodeOptions& options)
+    : node_{ std::make_shared<rclcpp::Node>("mtc_node", options) }
+    {
+    }
+
+    void MTCTaskNode::setupPlanningScene()
+    {
+    moveit_msgs::msg::CollisionObject object;
+    object.id = "object";
+    object.header.frame_id = "world";
+    object.primitives.resize(1);
+    object.primitives[0].type = shape_msgs::msg::SolidPrimitive::CYLINDER;
+    object.primitives[0].dimensions = { 0.1, 0.02 };
+
+    geometry_msgs::msg::Pose pose;
+    pose.position.x = 0.5;
+    pose.position.y = -0.25;
+    object.pose = pose;
+
+    moveit::planning_interface::PlanningSceneInterface psi;
+    psi.applyCollisionObject(object);
+    }
+
+    void MTCTaskNode::doTask()
+    {
+    task_ = createTask();
+
+    try
+    {
+        task_.init();
+    }
+    catch (moveit::task_constructor::InitStageException& e)
+    {
+        RCLCPP_ERROR_STREAM(LOGGER, e);
+        return;
+    }
+
+    if (!task_.plan(5))
+    {
+        RCLCPP_ERROR_STREAM(LOGGER, "Task planning failed");
+        return;
+    }
+    task_.introspection().publishSolution(*task_.solutions().front());
+
+    auto result = task_.execute(*task_.solutions().front());
+    if (result.val != moveit_msgs::msg::MoveItErrorCodes::SUCCESS)
+    {
+        RCLCPP_ERROR_STREAM(LOGGER, "Task execution failed");
+        return;
+    }
+
+    return;
+    }
+
+    moveit::task_constructor::Task MTCTaskNode::createTask()
+    {
+    moveit::task_constructor::Task task;
+    task.stages()->setName("demo task");
+    task.loadRobotModel(node_);
+
+    const auto& arm_group_name = "panda_arm";
+    const auto& eef_name = "hand";
+    const auto& hand_group_name = "hand";
+    const auto& hand_frame = "panda_hand";
+
+    // Set task properties
+    task.setProperty("group", arm_group_name);
+    task.setProperty("eef", eef_name);
+    task.setProperty("hand", hand_group_name);
+    task.setProperty("hand_grasping_frame", hand_frame);
+    task.setProperty("ik_frame", hand_frame);
+
+    moveit::task_constructor::Stage* current_state_ptr = nullptr;  // Forward current_state on to grasp pose generator
```

Comment:
Some of these lines are too long. I guess you could copy a clang-formatted version, possibly with a shorter line length.
Also, I wouldn't be against doing something like `namespace mtc = moveit::task_constructor`